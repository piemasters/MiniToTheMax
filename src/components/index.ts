export * from './Badge';
export * from './Badges';
export * from './CoverCategory';
export * from './CoverImage';
export * from './Footer';
export * from './Header';
export * from './PageLink';
export * from './Pagination';
export * from './Paint';
export * from './PostSummary';
export * from './PostTag';
export * from './Seo';
export * from './SimplePagination';
export * from './Video';
export * from './stateful';
