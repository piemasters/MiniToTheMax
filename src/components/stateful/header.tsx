import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import PureHeader from '../header';

interface NavLink {
  name: string;
  url: string;
}

const Header = () => {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
        }
      }
      file(name: { eq: "logo" }) {
        size
        relativeDirectory
        publicURL
      }
    }
  `);

  const pages: NavLink[] = [
    { name: 'Home', url: '/' },
    { name: 'Blog', url: '/blog' },
    { name: 'Showcase', url: '/showcase' },
    { name: 'Backlog', url: '/backlog' },
  ];

  return (
    <PureHeader
      title={data.site.siteMetadata.title}
      logo={data.file.publicURL}
      pages={pages}
    />
  );
};

export default Header;
