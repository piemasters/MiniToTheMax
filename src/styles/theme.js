export default {
  colors: {
    text: '#000',
    muted: '#999',
  },
};
