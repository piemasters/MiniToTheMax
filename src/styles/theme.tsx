export interface Theme {
  colors: {
    text: string;
    muted: string;
  };
}

const appTheme: Theme = {
  colors: {
    text: '#000',
    muted: '#999',
  },
};

export default appTheme;
