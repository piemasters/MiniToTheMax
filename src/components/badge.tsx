import React from 'react';
import { Theme } from '../styles/theme';
import { css, useTheme } from '@emotion/react';
import { BadgeNames } from '../types/app.types';
import PageLink from './page-link';
import kebabCase from 'lodash.kebabcase';
import tutorials from '../../content/assets/images/tutorials-badge.png';
import showcase from '../../content/assets/images/showcase-badge.png';
import battleReports from '../../content/assets/images/battle-reports-badge.png';
import reviews from '../../content/assets/images/reviews-badge.png';
import build from '../../content/assets/images/build-badge.png';

const Badge = ({ type }: { type: BadgeNames }): JSX.Element => {
  const theme = useTheme() as Theme;

  const badgeImages: { [key: string]: string } = {
    [BadgeNames.showcase]: showcase,
    [BadgeNames.tutorials]: tutorials,
    [BadgeNames.reviews]: reviews,
    [BadgeNames.battleReports]: battleReports,
    [BadgeNames.build]: build,
  };

  const linkStyles = css`
    color: ${theme.colors.textSecondary};
    display: flex;
    text-decoration: none;
    flex-direction: column;
    opacity: 0.9;
    &:hover {
      opacity: 1;
    }
  `;

  const badgeStyles = css`
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    width: 9rem; // 7rem for 5
  `;

  const badgeImageStyles = css`
    margin: 0 auto 0.4rem;
    width: 75%;
    opacity: 0.9;
    &:hover {
      opacity: 1;
    }
  `;

  const badgeTextStyles = css`
    text-align: center;
    text-transform: capitalize;
  `;

  return (
    <div css={badgeStyles} data-testid="badge">
      <PageLink
        to={kebabCase(type)}
        type={'cover'}
        direction={'up'}
        linkStyle={linkStyles}
      >
        <img src={badgeImages[type]} alt="" css={badgeImageStyles} />
        <p css={badgeTextStyles}>{type.replace(/([A-Z])/g, ' $1').trim()}</p>
      </PageLink>
    </div>
  );
};

export default Badge;
