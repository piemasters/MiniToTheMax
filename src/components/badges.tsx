import React from 'react';
import { Theme } from '../styles/theme';
import { useTheme } from 'emotion-theming';
import { css } from '@emotion/core';
import Badge from './badge';
import { BadgeNames } from '../types/app.types';

const Badges = () => {
  const theme: Theme = useTheme();

  const badgeContainerStyles = css`
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    justify-content: center;
  `;

  const badges: BadgeNames[] = [
    BadgeNames.showcase,
    BadgeNames.tutorials,
    BadgeNames.reviews,
    BadgeNames.battleReports,
    //BadgeNames.build,
  ];

  return (
    <div css={badgeContainerStyles} data-testid="badges">
      {badges.map((badge: BadgeNames) => {
        return <Badge type={badge} key={badge} />;
      })}
    </div>
  );
};

export default Badges;
