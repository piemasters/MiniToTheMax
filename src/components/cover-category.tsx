import React from 'react';
import PageLink from './page-link';
import { FluidObject } from 'gatsby-image';
import CoverImage from './cover-image';
import { Theme } from '../styles/theme';
import { css, useTheme } from '@emotion/react';

interface CoverCategoryProps {
  slug: string;
  title: string;
  img: FluidObject;
}

const CoverCategory = (category: CoverCategoryProps) => {
  const theme = useTheme() as Theme;
  const postStyle = css`
    margin: 1rem 0;
    transition: 0.3s;
    &:hover {
      box-shadow: 6px 6px 60px 5px ${theme.colors.backgroundTransparent};
    }
  `;

  const removeCategoryName = (title: string) => {
    if (title.slice(0, 15) === 'Battle Report: ') {
      return title.slice(15);
    } else if (title.slice(0, 8) === 'Review: ') {
      return title.slice(8);
    } else if (title.slice(0, 10) === 'Showcase: ') {
      return title.slice(10);
    } else if (title.slice(0, 10) === 'Tutorial: ') {
      return title.slice(10);
    } else if (title.slice(0, 19) === 'Review & Tutorial: ') {
      return title.slice(19);
    } else {
      return title;
    }
  };

  return (
    <div css={postStyle} data-testid="cover-category">
      <PageLink to={category.slug} type={'cover'} direction={'up'}>
        <CoverImage
          image={category.img}
          title={removeCategoryName(category.title)}
        />
      </PageLink>
    </div>
  );
};

export default CoverCategory;
