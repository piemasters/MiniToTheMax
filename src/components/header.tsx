import React from 'react';
import { graphql, useStaticQuery } from 'gatsby';
import { useTheme } from 'emotion-theming';
import { css } from '@emotion/core';
import PageLink from './page-link';
import { Theme } from '../styles/theme';

interface NavLink {
  name: string;
  url: string;
}

interface HeaderProps {
  title: string;
  logo: string;
  pages: NavLink[];
}

export const PureHeader = ({ title, logo, pages }: HeaderProps) => {
  const theme: Theme = useTheme();

  const NavHeader = css`
    padding: 1rem 0 3rem;
  `;

  const NavTitle = css`
    a {
      color: ${theme.colors.text};
      font-size: 3rem;
      text-decoration: none;
    }
  `;

  const NavList = css`
    display: flex;
    list-style-type: none;
    margin: 0;
  `;

  const NavItem = css`
    a {
      color: ${theme.colors.muted};
      font-size: 0.9rem;
      margin-right: 1.3rem;
      text-decoration: none;
      &:hover {
        color: #666666;
      }
    }
  `;

  return (
    <div css={NavHeader} data-testid="nav-header">
      <nav>
        <h1>
          <div css={NavTitle}>
            <PageLink type={'paintDrip'} to={'/'}>
              <img src={logo} alt="Logo" width={60 + 'px'} />
              {title}
            </PageLink>
          </div>
        </h1>
        <div css={NavList}>
          {pages.map((link: NavLink) => {
            return (
              <div css={NavItem} key={link.name}>
                <PageLink
                  type={'cover'}
                  linkActiveStyle={{ color: '#333333' }}
                  to={link.url}
                >
                  {link.name}
                </PageLink>
              </div>
            );
          })}
        </div>
      </nav>
    </div>
  );
};

const Header = () => {
  const data = useStaticQuery(graphql`
    query {
      site {
        siteMetadata {
          title
        }
      }
      file(name: { eq: "logo" }) {
        size
        relativeDirectory
        publicURL
      }
    }
  `);

  const pages: NavLink[] = [
    { name: 'Home', url: '/' },
    { name: 'Blog', url: '/blog' },
    { name: 'Showcase', url: '/showcase' },
    { name: 'Backlog', url: '/backlog' },
  ];

  return (
    <PureHeader
      title={data.site.siteMetadata.title}
      logo={data.file.publicURL}
      pages={pages}
    />
  );
};

export default Header;
