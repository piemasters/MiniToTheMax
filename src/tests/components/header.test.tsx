import React from 'react';
import { PureHeader as Header } from '../../components/header';
import { render, getByTestId } from '@testing-library/react';
import { ThemeProvider } from 'emotion-theming';
import theme from '../../styles/theme';

describe('Header', () => {
  test('renders correctly', () => {
    const data = {
      site: {
        siteMetadata: {
          title: 'MiniToTheMax',
        },
      },
      file: {
        publicURL: '/static/logo-37f736b45fa7360612c71ff66ce83af8.png',
      },
    };

    const pages = [
      { name: 'Home', url: '/' },
      { name: 'Blog', url: '/blog' },
      { name: 'Showcase', url: '/showcase' },
      { name: 'Backlog', url: '/backlog' },
    ];

    const { container } = render(
      <ThemeProvider theme={theme}>
        <Header
          title={data.site.siteMetadata.title}
          logo={data.file.publicURL}
          pages={pages}
        />
      </ThemeProvider>
    );

    expect(getByTestId(container, 'nav-header')).toHaveTextContent(
      'MiniToTheMax'
    );

    expect(container.querySelector(`header`)).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });
});
