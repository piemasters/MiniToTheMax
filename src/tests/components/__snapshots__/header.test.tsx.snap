// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders correctly 1`] = `
<div>
  <div
    css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    data-testid="nav-header"
  >
    <nav
      css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
    >
      <a
        aria-current="page"
        class=""
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        data-testid="page-link-paint"
        hex="#eb1d23"
        href="/"
      >
        <img
          alt="Logo"
          src="/static/logo-37f736b45fa7360612c71ff66ce83af8.png"
          width="60px"
        />
        <h1
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
        >
          MiniToTheMax
        </h1>
      </a>
      <div
        css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
      >
        <a
          aria-current="page"
          bg="#eb1d23"
          class=""
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          data-testid="page-link-cover"
          direction="up"
          duration="1"
          href="/"
          style="color: rgb(235, 29, 35);"
        >
          Home
        </a>
        <a
          bg="#eb1d23"
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          data-testid="page-link-cover"
          direction="up"
          duration="1"
          href="/blog"
        >
          Blog
        </a>
        <a
          bg="#eb1d23"
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          data-testid="page-link-cover"
          direction="up"
          duration="1"
          href="/showcase"
        >
          Showcase
        </a>
        <a
          bg="#eb1d23"
          css="You have tried to stringify object returned from \`css\` function. It isn't supposed to be used directly (e.g. as value of the \`className\` prop), but rather handed to emotion so it can handle it (e.g. as value of \`css\` prop)."
          data-testid="page-link-cover"
          direction="up"
          duration="1"
          href="/backlog"
        >
          Backlog
        </a>
      </div>
    </nav>
  </div>
</div>
`;
